src/components/Modals/Settings/Languages.tsx

```diff
@@ -0,0 +1,99 @@
+import { ArrowBackIcon } from '@chakra-ui/icons'
+import { Button, Flex, Icon, IconButton, ModalBody, ModalHeader } from '@chakra-ui/react'
+import { FaCheck } from 'react-icons/fa'
+import { useTranslate } from 'react-polyglot'
+import { RouteComponentProps, useHistory } from 'react-router-dom'
+import { RawText } from 'components/Text'
+import { preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectSelectedLocale } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { SettingsRoutes } from './Settings'
+
+/**
+ * Probably needs a better home
+ */
+export const locales = [
+  {
+    key: 'en',
+    label: 'English'
+  },
+  {
+    key: 'es',
+    label: 'Español'
+  },
+  {
+    key: 'fr',
+    label: 'français'
+  },
+  {
+    key: 'id',
+    label: 'Bahasa Indonesia'
+  },
+  {
+    key: 'ko',
+    label: '한국어'
+  },
+  {
+    key: 'pt',
+    label: 'Português'
+  },
+  {
+    key: 'ru',
+    label: 'Русский язык'
+  },
+  {
+    key: 'zh',
+    label: '中文'
+  }
+]
+
+export const Languages = (props: RouteComponentProps) => {
+  const dispatch = useAppDispatch()
+  const history = useHistory()
+  const selectedLocale = useAppSelector(selectSelectedLocale)
+  const translate = useTranslate()
+  const otherLocales = locales.filter(l => l.key !== selectedLocale)
+  const back = () => history.push(SettingsRoutes.Index)
```

Comment:
However, the A/C states that `Once a language is selected the user can return to the initial settings modal by clicking the back button on the top left of the modal` - so we shouldn't ever need this at all, since there's no need to be brought back to the previous route.